rpcq/_server.py

```diff
@@ -102,8 +167,12 @@ def exit_handler(self, f: Callable):
                     # empty_frame may either be:
                     # 1. a single null frame if the client is a REQ socket
                     # 2. an empty list (ie. no frames) if the client is a DEALER socket
-                    identity, *empty_frame, msg = done.result()
+                    identity, *empty_frame, msg, client_key = done.result()
                     request = from_msgpack(msg)
+                    try:
+                        request.params['client_key'] = client_key
+                    except Exception as e:
```

Comment:
I left this comment on the other PR accidentally. Repeating here: can this exception be more specific with `KeyError`?

Reply:
We may be able to remove this try/except altogether - it doesn't catch any 'expected' exceptions anymore, even a KeyError would be unexpected. `request.params` should always be defined and always a `dict`, and client_key will always be defined, AFAICT. What do you think?

Reply:
Agree, remove it 👍